daemons/attrd/attrd_cib.c

```diff
@@ -113,7 +114,7 @@ drop_values_in_deletion(xmlNode *xml, void *data)
         attrd_drop_removed_set(set_type, set_id);
 
     } else if (!pcmk__str_empty(node_id)) {
-        attrd_drop_removed_values(node_id);
+        attrd_drop_removed_values(node_id, GPOINTER_TO_INT(data));
```

Comment:
Another thing that's not obvious just by inspection, is why we pass the peer version (and drop immediately if applicable) only when a `<node_state>` or `<transient_attributes>` element is removed. We don't do this for a single set or a single attribute.

I understand what we're doing here (I think). "`<node_state>` or `<transient_attributes>` was removed" is used as a necessary and sufficient condition for "the node left the cluster." That relationship is only implicit though.